app/src/main/java/nerd/tuxmobil/fahrplan/congress/base/SessionsAdapter.kt

```diff
@@ -0,0 +1,232 @@
+    numDays: Int,
+    useDeviceTimeZone: Boolean
+) : ArrayAdapter<Session?>(
+    context!!, layout, list!!
+) {
+    companion object {
+        private const val TYPE_ITEM = 0
+        private const val TYPE_SEPARATOR = 1
+        private const val NUM_VIEW_TYPES = 2
+    }
+
+    @get:JvmName("getAdapterContext")
+    protected var context: Context? = null
+        private set
+
+    private val list: List<Session>?
+    private val numDays: Int
+    private var mMapper: ArrayList<Int>? = null
+    private var mSeparatorStrings: ArrayList<String>? = null
+    private var mSeparatorsSet: TreeSet<Int>? = null
+    protected val useDeviceTimeZone: Boolean
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        val rowView: View
+        var viewHolder: ViewHolder? = null
+        var viewHolderSeparator: ViewHolderSeparator? = null
+        val type = getItemViewType(position)
+
+        if (convertView == null) {
+            // clone the inflater using the ContextThemeWrapper
+            val inflater = context!!.getLayoutInflater()
+            val localInflater = inflater.cloneInContext(context)
+
+            when (type) {
+                TYPE_ITEM -> {
+                    rowView = localInflater.inflate(R.layout.session_list_item, parent, false)
+                    viewHolder = ViewHolder(
+                        rowView.requireViewByIdCompat(R.id.session_list_item_title_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_subtitle_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_speakers_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_language_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_day_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_time_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_room_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_duration_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_video_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_no_video_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_without_video_recording_view),
+                    )
+                    rowView.tag = viewHolder
+                }
+                TYPE_SEPARATOR -> {
+                    rowView = localInflater.inflate(R.layout.session_list_separator, parent, false)
+                    viewHolderSeparator = ViewHolderSeparator()
+                    viewHolderSeparator.text = rowView.requireViewByIdCompat(R.id.session_list_separator_title_view)
+                    rowView.tag = viewHolderSeparator
+                }
+                else -> {
+                    throw IllegalStateException("Unknown type: $type")
```

Comment:
The `throw` statement can be replaced with the `error("Unknown type: $type")` convenience function here and below.